frontend/src/components/Customer/CustomerSignUp.tsx

```diff
@@ -0,0 +1,358 @@
+      rows: 1,
+      cols: 1,
+      title: 'Apple Krumb'
+    },
+    {
+      img: require('../../assets/bavarian_kreme.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Bavarian Kreme'
+    },
+    {
+      img: require('../../assets/blueberry.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Blueberry'
+    },
+    {
+      img: require('../../assets/boston_kreme.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Boston Kreme'
+    },
+    {
+      img: require('../../assets/chocolate_frosted.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Chocolate Frosted'
+    },
+    {
+      img: require('../../assets/chocolate_glaze.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Chocolate Glaze'
+    },
+    {
+      img: require('../../assets/marble_frosted.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Marble Frosted'
+    },
+    {
+      img: require('../../assets/cinnamon_sugar.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Cinnamon Sugar'
+    },
+    {
+      img: require('../../assets/old_fashioned.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Old Fashioned'
+    },
+    {
+      img: require('../../assets/original_glaze.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Original Glaze'
+    },
+    {
+      img: require('../../assets/powdered_sugar.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Powdered Sugar'
+    },
+    {
+      img: require('../../assets/cruller.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Cruller'
+    },
+    {
+      img: require('../../assets/sour_cream.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Sour Cream'
+    },
+    {
+      img: require('../../assets/strawberry_frosted.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Strawberry Frosted'
+    },
+    {
+      img: require('../../assets/vanilla_frosted.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Vanilla Frosted'
+    },
+    {
+      img: require('../../assets/coconut.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Coconut'
+    },
+    {
+      img: require('../../assets/jelly.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Jelly'
+    },
+    {
+      img: require('../../assets/chocolate_kreme.jpeg'),
+      rows: 1,
+      cols: 1,
+      title: 'Chocolate Kreme'
+    }
+  ];
+function Copyright() {
+    return (
+      <Typography variant="body2" color="text.secondary" align="center">
+        {'Copyright © '}
+        <Link color="inherit" to="https://mui.com/">
+          Dronuts Group 3
+        </Link>{' '}
+        {new Date().getFullYear()}
+        {'.'}
+      </Typography>
+    );
+  }
+const theme = createTheme();
+
+export interface newCustomer {
+    first_name: string,
+    last_name: string,
+    email: string,
+    phone_number: string, 
+    password: string,
+    address: Array<LocationInterface>
+}
+
+
+
+export function CustomerSignUp() {
+    
+    const [customerId, setCustomerId] = React.useState(0);
```

Comment:
I would recommend introducing a CustomerInterface similarly to OrderInterface or DonutInterface, and useState on the one object. Then this is passed through to backend API, and updated in form.